src/00/02-introspect/proposal.adoc

```diff
@@ -0,0 +1,879 @@
+interoperability to work, the tool consumers and target tools must operate on
+this minimal pair of functions to bootstrap their interactions. To make that
+possible, this design follows some basic tenets:
+
+[horizontal]
+
+Minimal:: The interface of the target tool is a single universal command line
+argument for each of the two operations.
+
+Concise:: The information communicated to and from the target tool and consumer
+is as brief as needed to convey the required information.
+
+Robust:: The interface and information should not result in failure conditions
+for either the consumer or target tool. Both ends of the interactions need to
+rely on the stability of the interface to then be able to interoperate.
+
+=== Introspection
+
+The consumer can use a single method to query the target tool and obtain all the
+capabilities that are available or specifically requested. The use case
+supported is for unbounded _introspection_ of the available capabilities with a
+single valueless `--std-info` option.
+
+And unbounded introspection simply returns everything the tool is capable
+of doing. The tool has the option to respond with either all minimal single
+(aka bare) versions or full version ranges. Either can be trivially implemented
+by tools as most time it can be a hard-wired response text.
+
+Running a tool with the option would look like the following:
+
+[source,shell]
+----
+$ tool --std-info
+----
+
+And could produce this as a minimal JSON output to indicate the single version
+of the capabilities it supports:
+
+[source,json]
+----
+include::example-01.json[]
+----
+
+Or could produce this as a JSON output in the case of full version ranges:
+
+[source,json]
+----
+include::example-02.json[]
+----
+
+Which would minimally indicate that the tool only supports the introspection
+capability at versions "1.0.0" through "2.5.0".
+
+Per the findings of the <<User Interface>> research and the consensus of the
+<<SG15: P2717R2 (2023-06-16)>> polling a tool can, additionally, provide the
+introspection information in a file accompanying the tool. There is a challenge
+when providing such an introspection file though: It is not practical to specify
+an absolute location, or locations, across the variety of operating systems and
+tools in the programming ecosystem. As such we provide some possibilities:
+
+1. The name of an introspection file will be the name of the top level invoked
+tool executable (or script, or equivalent) with any type extension
+(i.e. "`.exe`") removed if it exists. That base name will be appended with the
+`.stdinfo` text. For example: `cl.exe` => `cl.stdinfo`, or
+`pass:[g++]` => `pass:[g++].stdinfo`.
+
+2. The introspection file can be found: in the same filesystem location
+as the tool executable (or script, or equivalent), in an implementation defined
+location relative to the tool location, or in an implementation defined global
+location (i.e. an absolute path location).
```

Comment:
This should have a sublist so it is obvious which is used.

```suggestion
2. The introspection file can be found: 
.. in the same filesystem location as the tool executable (or script, or equivalent)
.. in an implementation defined location relative to the tool location
.. in an implementation defined global location (i.e. an absolute path location)
```

Furthermore not having a central location as defined by at least an environment variable separate from the implementation is not going to turn out well. Developers will always want to override or provide information on a tool themselves and having a central location like e.g., `pkg-config` is worthwhile.